packages/ecs-agent/ecs.config

```diff
@@ -1,2 +1,6 @@
 ECS_LOGFILE=/var/log/ecs/ecs-agent.log
 ECS_LOGLEVEL="{{settings.ecs.loglevel}}"
+{{#if settings.network.http-proxy}}
+HTTP_PROXY={{settings.network.http-proxy}}
+{{/if}}
+NO_PROXY={{#each settings.network.no-proxy}}{{#if @index}},{{/if}}{{this}}{{/each}}{{#if settings.network.no-proxy}},{{/if}}169.254.169.254,169.254.170.2,/var/run/docker.sock
```

Comment:
Separately - I think handlebars has `@first` and `@last` in addition to `@index`, so you might be able to clarify why we're checking `@index` by using `@first` instead, and instead of having a separate `#if settings.network.no-proxy`, use `@last`?  (same in proxy-env)

Reply:
Oh I didn't know `@first` and `@last` are a thing.. that's such a weird construct. Thanks!

Reply:
For `proxy-env` it's slightly different because I cannot guarantee I'll be adding additional hosts to the list. The additional elements are added on the condition that `settings.kubernetes.api-server` is set (meaning it's running kubelet) since only kubernetes needs them.